File: src/main/java/com/drunkensoftball/api/auth/google/repo/GoogleAuthenticationRepository.java
package com.drunkensoftball.api.auth.google.repo;

import com.drunkensoftball.api.auth.google.domain.GoogleAuthentication;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface GoogleAuthenticationRepository extends JpaRepository<GoogleAuthentication, Long> {

    Optional<GoogleAuthentication> findByGoogleId(String googleId);

}

File: src/main/kotlin/com/drunkensoftball/api/auth/repo/GoogleAuthenticationRepository.kt
package com.drunkensoftball.api.auth.repo

import com.drunkensoftball.api.auth.domain.GoogleAuthentication
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface GoogleAuthenticationRepository : JpaRepository<GoogleAuthentication, Long> {

    fun findByGoogleId(googleId: String): Optional<GoogleAuthentication>
}
